src/EditorFeatures/CSharpTest/ConvertNamespace/ConvertNamespaceRefactoringTests.cs

```diff
@@ -0,0 +1,911 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeStyle;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.CodeStyle;
+using Microsoft.CodeAnalysis.CSharp.ConvertNamespace;
+using Microsoft.CodeAnalysis.Editor.UnitTests.CodeActions;
+using Microsoft.CodeAnalysis.Testing;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.ConvertNamespace
+{
+    using VerifyCS = CSharpCodeRefactoringVerifier<ConvertNamespaceCodeRefactoringProvider>;
+
+    public class ConvertNamespaceRefactoringTests
+    {
+        #region Convert To File Scoped
+
+        [Fact]
+        public async Task TestNoConvertToFileScopedInCSharp9()
+        {
+            var code = @"
+namespace $$N
+{
+}
+";
+            await new VerifyCS.Test
+            {
+                TestCode = code,
+                FixedCode = code,
+                LanguageVersion = LanguageVersion.CSharp9,
+                Options =
+                {
+                    { CSharpCodeStyleOptions.NamespaceDeclarations, NamespaceDeclarationPreference.BlockScoped }
+                }
+            }.RunAsync();
+        }
+
+        [Fact]
+        public async Task TestNoConvertToFileScopedInCSharp10WithFileScopedPreference()
+        {
+            var code = @"
+namespace $$N
+{
+}
+";
+            await new VerifyCS.Test
+            {
+                TestCode = code,
+                FixedCode = code,
+                LanguageVersion = LanguageVersion.CSharp10,
+                Options =
+                {
+                    { CSharpCodeStyleOptions.NamespaceDeclarations, NamespaceDeclarationPreference.FileScoped }
+                }
+            }.RunAsync();
+        }
+
+        [Fact]
+        public async Task TestConvertToFileScopedInCSharp10WithBlockScopedPreference()
+        {
+            await new VerifyCS.Test
+            {
+                TestCode = @"
+namespace $$N
+{
+}
+",
+                FixedCode = @"
+namespace $$N;
+",
+                LanguageVersion = LanguageVersion.CSharp10,
+                Options =
+                {
+                    { CSharpCodeStyleOptions.NamespaceDeclarations, NamespaceDeclarationPreference.BlockScoped }
```

Comment:
This is more confusing.. if the preference is block scoped why are we offering to change to file scoped? Do we only offer when the existing code matches the preference?

ohhh wait.. is this because if the code doesn't match the preference, the fixer does the work? that probably makes sense..

Reply:
Yup. This is how 'use expression body' works as well.